02-intermediate-ruby/advanced-variables.md

```diff
@@ -15,137 +15,296 @@
 |:----------------:|:------------------:|------------------------------------------|
 |Local variable    | **a-z** *or* **_** | Available only within the immediate scope.
 |Instance variable | **@**              | Available to a specific instance of a class
-|Class variable    | **@@**             | Available anywhere in a specific class (including all instances) (avoid me)
-|Global variable   | **$**              | Available everywhere (don't use me).
+|Class variable    | **@@**             | Available anywhere in a specific class (including all instances) (avoid using these)
+|Global variable   | **$**              | Available everywhere (don't use these)
 |Constant          | **ALL_CAPS**       | Available in the scope of their definition
 
 ### Local Variable
-- Local variables start with a lowercase letter or an underscore.
-- Local variables are only available within the block of its initialization.
+- Local variables start with a lowercase letter or an underscore
+- Local variables are only available within the block of their initialization
 - Local variables will raise an error if they are read before they're created
 - Local variables are used often
 
+Let's examine the following code:
 ```ruby
-my_name = "<3 Jeremy!"
+class Library
+  def open_library
+    message = "The library is open"
+    puts message
+  end
+
+  def close_library
+    message = "The library is closed"
```

Comment:
It might be good to use a different variable name here so there isn't any confusion between sharing variables across methods